### DISCUSSION TITLE
IPv6 on STM32?

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#10289] [STM32 / Pyboard]
Does the STM32 port support IPv6? The docs say IPv6 support is dependent on the port, but doesn't lead to a list of ports which support it. Is it something that needs to be compiled in?

**Comment by jimmo** [comment]
@victorallume Sorry about the delay responding.

The STM32 port does not currently support IPv6 (it's disabled via build options).

However, it's easy to enable this in LWIP, and off the top of my head I don't think there are any major show stoppers for enabling this in MicroPython itself. The main limitation is support for configuring the network interface etc, and there is current work ongoing to add this for the CYW43 driver used in the PYBD (and Portenta, and Pico W). I'm not sure what the ETA will be though.

**Comment by victorallume** [comment]
I see that IPv6 support for cyw43/rp2 is merged https://github.com/micropython/micropython/pull/9108 - what would be required to get support for the PYBD (which uses the same wifi chip)?
